### PR TITLE
[C++ Worker]Add api get placement group

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Add api: GetAllPlacementGroups, GetPlacementGroup, GetPlacementGroupById

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

File: cpp/include/ray/api.h [hunk 1]
```diff
@@ -150,6 +150,19 @@ PlacementGroup CreatePlacementGroup(
...
+/// \param[in] id Id of the placement group.
+/// \param[in] timeout_seconds Timeout in seconds.
+/// \return True if the placement group is created. False otherwise.
+bool WaitPlacementGroupReady(const std::string &id, int timeout_seconds);
+
+std::vector<PlacementGroup> GetAllPlacementGroups();
+
+PlacementGroup GetPlacementGroupById(const std::string &id);
+PlacementGroup GetPlacementGroup(const std::string &name);
+PlacementGroup GetGlobalPlacementGroup(const std::string &name);
```
Comment (SongGuyang) [inline cpp/include/ray/api.h:164]: Can we also add some comments for all the new API to users?

File: cpp/include/ray/api/task_options.h [hunk 2]
```diff
@@ -45,11 +45,19 @@ enum class PlacementStrategy {
...
+  REMOVED = 2,
+  RESCHEDULING = 3,
+};
+
 struct PlacementGroupCreationOptions {
   bool global;
   std::string name;
   std::vector<std::unordered_map<std::string, double>> bundles;
   PlacementStrategy strategy;
+  PlacementGroupState state;
```
Comment (SongGuyang) [inline cpp/include/ray/api/task_options.h:60]: Why add this state to `PlacementGroupCreationOptions `? It's creation options, and I think we can't add this state.

Comment (raulchen) [reply cpp/include/ray/api/task_options.h:63]: Right, the state should be internal

File: cpp/include/ray/api.h [hunk 3]
```diff
@@ -128,28 +128,36 @@ inline void ExitActor() { ray::internal::GetRayRuntime()->ExitActor(); }
...
 ///
 /// \param[in] placement_group_id Id of the placement group.
 void RemovePlacementGroup(const std::string &placement_group_id);
 
+/// Wait for the placement group to be ready within the specified time.
+///
+/// \param[in] id Id of the placement group.
+/// \param[in] timeout_seconds Timeout in seconds.
+/// \return True if the placement group is created. False otherwise.
+bool WaitPlacementGroupReady(const std::string &id, int timeout_seconds);
```
Comment (raulchen) [inline cpp/include/ray/api.h:148]: This is not needed. Users should use `pg.Wait()`

**Review by raulchen — APPROVED** [review]

**Pull request merged**